### DISCUSSION TITLE
How to use Phlex components with TurboFrame?

### DISCUSSION THREAD

**Discussion by shetty-tejas** [yippee-fun/phlex#664] [Help] [answered]
Are there any documentation or live code examples showcasing effective integration of Turbo Frame and Turbo Stream with Phlex? While I managed to make Phlex work with Turbo Stream, I'm uncertain if my solution is optimal. However, I'm still not able to work with Turbo Frame.

Either ways, those who have successfully incorporated both into their applications, could you provide some code examples and/or configuration settings? In my opinion, a thread with some code examples can be of huge help to anyone that's new to Phlex!

**Comment by mtomov** [comment, marked as answer]
We're using them without issues.

<img width="656" alt="image" src="https://github.com/phlex-ruby/phlex/assets/1651750/4471d867-907d-49a6-8923-8f5a136313b2">


```ruby

  include Phlex::Rails::Helpers::TurboStream
  include Phlex::Rails::Helpers::TurboStreamFrom
  include Phlex::Rails::Helpers::TurboFrameTag

```

just include those 3 in ApplicationComponent and that's it, no other changes

then as usual `turbo_stream_from(@recording)` ..

**Reply by shetty-tejas** [reply]
One of the places that I'm not able to work around is having turbo_stream.erb files. How can I have something like this, without creating an erb file for the same though? And more importantly, how to use these "partials" with format.turbo_stream in a respond_to block?

**Reply by olance** [reply]
@shettytejas did you find a way?

**Reply by shetty-tejas** [reply]
@olance Not quite :) Still waiting on a solution!

**Reply by olance** [reply]
I have found [this](https://github.com/orgs/phlex-ruby/discussions/303#discussioncomment-8503236) in the meantime, have you tried? (I still have to)

**Reply by joeldrapper** [reply]
There's no need to use turbo_stream.erb files anymore. Instead, you can just render the component directly in the controller. See this example https://x.com/_swanson/status/1821256186844557593

Here's the PR where this feature was introduced in `turbo-rails`. https://github.com/hotwired/turbo-rails/pull/433
